### PR TITLE
Setuptools and slight reorganization

### PR DESCRIPTION
**Pull request by franp9am**
Adding a setup file to make it installable
Moving files into a charsplit/ subdirectory
Changing the data file to a json
Wrapping the split_compound function into a class Splitter
Renaming the char_split_train script to training
Editing Readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,29 +16,24 @@ A model is provided, trained on 1 Mio. German nouns from Wikipedia.
 ### Usage ###
 **Train** a new model:
 ```
```
Comment (Borda) [inline README.md:20]:
```suggestion
```bash
```

File: README.md [hunk 2]
```diff
@@ -16,29 +16,24 @@ A model is provided, trained on 1 Mio. German nouns from Wikipedia.
...
+training.py <your_train_file> <output_file>
 ```
-where `<your_train_file>` contains one word (noun) per line.
+from command line, where `<your_train_file>` contains one word (noun) per line.
 
 **Compound splitting**
 
-From command line:
+In python
 ```
```
Comment (Borda) [inline README.md]:
```suggestion
```python
```

File: charsplit/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from charsplit.splitter import Splitter
+
+__author__ = 'don.tuggener@gmail.com'
```
Comment (Borda) [inline charsplit/__init__.py]:
```suggestion
__author_email__ = 'don.tuggener@gmail.com'
```
you may add also `__author__` and `__version__` here :]

File: charsplit/splitter.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+Split German compound words
+"""
+
+from pathlib import Path
+import re
+import sys
+import json
+
+# path to the datafile
+NGRAM_PATH = Path(__file__).parent / "ngram_probs.json"
```
Comment (Borda) [inline charsplit/splitter.py]: move it to class `Splitter`

File: charsplit/splitter.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+            ngram_probs = self._get_ngram_probs()
+
+            # Extract all ngrams
+            for k in range(len(word)+1, 2, -1):
+
+                # Probability of first compound, given by its ending prob
+                if pre_slice_prob == [] and k <= len(pre_slice):
+                    end_ngram = pre_slice[-k:]  # Look backwards
+                    pre_slice_prob.append(ngram_probs["suffix"].get(end_ngram, -1))    # Punish unlikely pre_slice end_ngram
```
Comment (Borda) [inline charsplit/splitter.py:53]:
```suggestion
                    # Punish unlikely pre_slice end_ngram
                    pre_slice_prob.append(ngram_probs["suffix"].get(end_ngram, -1))
```

File: README.md [hunk 6]
```diff
@@ -16,29 +16,24 @@ A model is provided, trained on 1 Mio. German nouns from Wikipedia.
 ### Usage ###
 **Train** a new model:
 ```
-python char_split_train.py <your_train_file>
+training.py <your_train_file> <output_file>
```
Comment (Borda) [inline README.md]:
```suggestion
training.py --input_file <your_train_file> --output_file <output_file>
```

File: charsplit/training.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    with open(args.output_file, "w") as f:
+        json.dump({
+            "prefix": start_ngrams,
+            "infix": in_ngrams,
+            "suffix": end_ngrams
+        }, f)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (Borda) [inline charsplit/training.py]: rather `args = parse_args()` here and pass to main just the parameters `input_file` and `output_file`

**Review by Borda — COMMENTED** [review]
just a few friendly suggestions... :]

**Comment by dtuggener** [comment]
Thanks for your suggestions, they have been merged.

**Pull request merged**